Jenkinsfile

```diff
@@ -66,28 +87,41 @@ pipeline {
                 }
               }
             }
-            stage('xCORE builds') {
-              steps {
-                // build all the supported firmware applications
-                runForEach(['i2c', 'i2c/host_xcore', 'spi', 'usb', 'xscope']) { app ->
-                  withTools(params.TOOLS_VERSION) { // the XTC tools are necessary to build the XSCOPE host application
-                    dir("${REPO}/examples/${app}") {
-                          sh 'cmake -G "Unix Makefiles" -B build'
-                          sh "xmake -C build"
-                    }
-                  }
-                }
-              }
+          }
+          post {
+            cleanup {
+              xcoreCleanSandbox()
             }
-            stage('Build documentation') {
+          }
+        }
+
+        stage('Build documentation') {
+          agent {
+            label 'documentation'
+          }
+          stages {
+            stage('Docs') {
               steps {
-                dir("${REPO}/${REPO}") {
-                  runXdoc('doc')
+                createVenv("requirements.txt")
```

Comment:
I prefer  https://github.com/xmosnotes/an02017/blob/073d9d1c05591f36eff9a7286b39bfe99975aa16/Jenkinsfile#L8
but also fine
`
def buildDocs(String zipFileName) {
    withVenv {
        sh 'pip install git+ssh://git@github.com/xmos/xmosdoc@v6.1.1'
        sh 'xmosdoc'
        zip zipFile: zipFileName, archive: true, dir: 'doc/_build'
    }
}


`

Reply:
I used a similar function after I cleaned it up a bit. The changes are in https://github.com/xmos/lib_device_control/pull/111/commits/ and https://github.com/xmos/lib_device_control/pull/111/commits/4b44aed7b7a20db44452529c37959aeb3acddc13.